### PR TITLE
odgi `viz`, `depth`, `degree`, and `stats` can work with not optimized, but compacted graphs

### PR DESCRIPTION
**Pull request by AndreaGuarracino**
This tackles the problem encountered here https://github.com/pangenome/odgi/pull/301

This could be extended also for odgi `sort`, `layout`, and `pathindex`, but it would require changes in the XP index.

### CODE REVIEW

File: src/subcommand/stats_main.cpp [hunk 1]
```diff
@@ -141,7 +141,14 @@ int main_stats(int argc, char** argv) {
 			utils::handle_gfa_odgi_input(infile, "stats", args::get(progress), num_threads, graph);
         }
     }
-    ///graph.display();
+
+    /// we need to check in advance if the graph is compacted
```
Comment (AndreaGuarracino) [inline src/subcommand/stats_main.cpp:145]: The compacted IDs are needed only where we use the `position_map` vector, that is when `mean_links_length` and/or `sum_of_path_node_distances` are computed.

**Review by subwaystation — APPROVED** [review]
Alright, I think this is ready to go. We can merge this, which will override some changes I made in https://github.com/pangenome/odgi/pull/301/. 
I looked again at `sort`, `layout`, and `pathindex`. The code in `xp.cpp` can become fairly complex and interesting to test, if we would go the same way as in this PR.
So for these 3 subcommands, I would suggest to keep the hard optimize definition avoiding downstream bugs. 

What do you think @AndreaGuarracino ?

File: src/subcommand/stats_main.cpp (hunk 1 above)
Comment (subwaystation) [reply src/subcommand/stats_main.cpp]: We need to have this check if we have `-y, --yaml`.

Comment (subwaystation) [reply src/subcommand/stats_main.cpp]: Let me push something.

Comment (subwaystation) [reply src/subcommand/stats_main.cpp]: Done.

**Comment by subwaystation** [comment]
@AndreaGuarracino Please merge, if you are happy with it :+1:

**Comment by AndreaGuarracino** [comment]
Prophecy: we will talk about this again, but for now the show must go on

**Comment by subwaystation** [comment]
> Prophecy: we will talk about this again, but for now the show must go on

Afraid so. Fingers crossed!

**Pull request merged**